src/hrana.ts

```diff
@@ -26,7 +26,13 @@ export function _createClient(config: ExpandedConfig): HranaClient {
         );
     }
     const url = encodeBaseUrl(scheme, config.authority, config.path);
-    const client = hrana.open(url, config.authToken);
+    
+    let client: hrana.Client;
+    try {
+      client = hrana.open(url, config.authToken);
+    } catch (e) {
+      throw new LibsqlError('The WebSocket (Hrana) client failed to open, try using "https" URL if your platform does not support web sockets', "HRANA_WEBSOCKET_ERROR");
+    }
```

Comment:
This `catch` is too general, it consumes all exceptions and hides the root cause. The `hrana.open()` call can fail due to many reasons, WebSockets being unsupported is just one of them.

Reply:
would it be better to put this in hrana-client like this
![image](https://user-images.githubusercontent.com/22666749/233941776-a1ae584e-a283-41b8-a7a5-a791012e2c46.png)

at least here it would isolate it to the WebSocket creation

Reply:
The solution that I came up with was as follows:
- Update `@libsql/isomorphic-ws` to export `undefined` when the `WebSocket` global is not found
- Update `@libsql/hrana-client` to throw an instance of `WebSocketUnsupportedError` when the `WebSocket` imported from `@libsql/isomorphic-ws` is `undefined`
- Update `@libsql/client` (this package) to throw an instance of `LibsqlError` with error code `WEBSOCKETS_NOT_SUPPORTED` when `hrana.open()` throws `WebSocketUnsupportedError`

The last step is implemented in PR #25, would you mind having a look whether the error message looks reasonable to you? :)